src/Adapter/PlatformServices.Desktop/Utilities/DeploymentUtility.cs

```diff
@@ -156,27 +144,37 @@ private bool Deploy(string source, IRunContext runContext, ITestExecutionRecorde
         /// <param name="deploymentItems">The deployment item.</param>
         /// <param name="testSource">The test source.</param>
         /// <param name="deploymentDirectory">The deployment directory.</param>
-        /// <param name="deploymentWarnings">Warnings.</param>
+        /// <param name="resultsDirectory">Root results directory</param>
+        /// <returns>Returns a list of deployment warnings</returns>
         [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes", Justification = "Requirement is to handle all kinds of user exceptions and message appropriately.")]
-        private void Deploy(IList<DeploymentItem> deploymentItems, string testSource, string deploymentDirectory, out IEnumerable<string> deploymentWarnings)
+        private IEnumerable<string> Deploy(IList<DeploymentItem> deploymentItems, string testSource, string deploymentDirectory, string resultsDirectory)
         {
-            Debug.Assert(!string.IsNullOrEmpty(deploymentDirectory), "Deployment directory is null/empty");
-            Debug.Assert(this.fileUtility.DoesDirectoryExist(deploymentDirectory), "Deployment directory " + deploymentDirectory + " does not exist");
-            Debug.Assert(!string.IsNullOrEmpty(testSource), "TestSource directory is null/empty");
-            Debug.Assert(this.fileUtility.DoesFileExist(testSource), "TestSource " + testSource + " does not exist.");
+            Validate.IsFalse(string.IsNullOrWhiteSpace(deploymentDirectory), "Deployment directory is null or empty");
```

Comment:
>Validate [](start = 12, length = 8)

Just thinking if we could reuse ValidateArg here.
Should we add tests for these validation cases as well ?

Reply:
It is a private function and hence not adding it.. ideally these arg validations are redundant as well. Will pick up later